lib/src/main/java/com/scalar/admin/kubernetes/Pauser.java

```diff
@@ -77,67 +94,155 @@ public PausedDuration pause(int pauseDuration, @Nullable Long maxPauseWaitTime)
+    // Run an unpause operation.
+    UnpauseFailedException unpauseFailedException = null;
+    try {
+      unpauseWithRetry(requestCoordinator, MAX_UNPAUSE_RETRY_COUNT);
+    } catch (Exception e) {
+      unpauseFailedException = new UnpauseFailedException("Unpause operation failed.", e);
+    }
 
+    // Get pods and deployment information after pause.
     TargetSnapshot targetAfterPause;
     try {
-      targetAfterPause = targetSelector.select();
+      targetAfterPause = getTarget();
     } catch (Exception e) {
-      throw new PauserException(
-          "Failed to find the target pods to examine if the targets pods were updated during"
-              + " paused.",
-          e);
+      PauserException pauserException =
+          new PauserException(GET_TARGET_AFTER_PAUSE_ERROR_MESSAGE, e);
+      if (unpauseFailedException == null) {
+        throw pauserException;
+      } else {
+        throw new UnpauseFailedException(UNPAUSE_ERROR_MESSAGE, pauserException);
+      }
     }
 
-    if (!target.getStatus().equals(targetAfterPause.getStatus())) {
-      throw new PauserException("The target pods were updated during paused. Please retry.");
+    // Check if pods and deployment information are the same between before pause and after pause.
+    boolean isTargetStatusEqual;
+    try {
+      isTargetStatusEqual = targetStatusEquals(targetBeforePause, targetAfterPause);
+    } catch (Exception e) {
+      StatusCheckFailedException statusCheckFailedException =
+          new StatusCheckFailedException(STATUS_CHECK_ERROR_MESSAGE, e);
+      if (unpauseFailedException == null) {
+        throw statusCheckFailedException;
+      } else {
+        throw new UnpauseFailedException(UNPAUSE_ERROR_MESSAGE, statusCheckFailedException);
+      }
     }
 
-    return new PausedDuration(startTime, endTime);
+    // Create an error message if any of the operations failed.
+    String errorMessage =
+        createErrorMessage(unpauseFailedException, pauseFailedException, isTargetStatusEqual);
+
+    // We use the exceptions as conditions instead of using boolean flags like `isPauseOk`, etc. If
+    // we use boolean flags, it might cause a bit large number of combinations. For example, if we
+    // have three flags, they generate 2^3 = 8 combinations. It also might make the nested if
+    // statement or a lot of branches of the switch statement. That's why we don't use status flags
+    // for now.
+
+    // Return the final result based on each process.
+    if (unpauseFailedException != null) { // Unpause Failed.
+      // Unpause issue is the most critical because it might cause system down.
+      throw new UnpauseFailedException(errorMessage, unpauseFailedException);
+    } else if (pauseFailedException != null) { // Pause Failed.
+      // Pause Failed is second priority because pause issue might be caused by configuration error.
+      throw new PauseFailedException(errorMessage, pauseFailedException);
```

Comment:
The first `PauseFailedException` instance contains a `PauseFailedException` instance as a cause. Is it expected? How about making `pauseFailedException` just Exception not `PauseFailedException`?

Reply:
@komamitsu 
Ah, yes. You're right. I think we can throw the `pauseFailedException ` as is instead of `new PauseFailedException()`.

I will fix it. I think we can fix this issue on the `buildException()` side that we are discussing in another comment.

Thank you for pointing it out!